src/ripple/core/impl/JobQueue.cpp

```diff
@@ -299,72 +248,94 @@ JobQueue::isStopped() const
     return stopped_;
 }
 
-void
-JobQueue::getNextJob(Job& job)
+Job
+JobQueue::getNextJob()
```

Comment:
Actually, when I was doing this refactor, I wanted to eliminate this function and turn it into a direct-invoked lambda, precisely because of the issue that you describe; it would prevent any possibility of abuse. I think I may do that.